Sources/NIO/Utilities.swift

```diff
@@ -130,22 +175,60 @@ public enum System {
     /// - returns: An array of network devices available on this machine.
     /// - throws: If an error is encountered while enumerating interfaces.
     public static func enumerateDevices() throws -> [NIONetworkDevice] {
+        var devices: [NIONetworkDevice] = []
+        devices.reserveCapacity(12)  // Arbitrary choice.
+
+#if os(Windows)
+        var ulSize: ULONG = 0
+        _ = GetAdaptersAddresses(ULONG(AF_UNSPEC), 0, nil, nil, &ulSize)
+
+        let alignment: Int = MemoryLayout<IP_ADAPTER_ADDRESSES>.alignment
+        let pBuffer: UnsafeMutableRawPointer =
```

Comment:
I think we should bind this pointer to the appropriate memory type here. This avoids the invalid use of `assumingMemoryBound` elsewhere in this code.

Reply:
I don't fully know how to `bindMemory(to:capacity:)` here, the `capacity:` being the bit that I cannot immediately see how to compute.

Reply:
The `capacity` is `Int(ulSize) / MemoryLayout<IP_ADAPTER_ADDRESSES>.stride`.

Reply:
Ah right - the capacity of the buffer is in terms of the `IP_ADAPTER_ADDRESSES`.  I was tripping myself up because of the linked-list aspect of traversal.  Thanks.

Reply:
I think we missed doing this.

Reply:
Ah, oops, fixed.